response_operations_ui/controllers/notify_controller.py

```diff
@@ -1,59 +1,67 @@
+import json
 import logging
 import structlog
-import requests
-
 from response_operations_ui.exceptions.exceptions import NotifyError
 from flask import current_app as app
-from requests.exceptions import HTTPError
-
+from google.cloud import pubsub_v1
 
 logger = structlog.wrap_logger(logging.getLogger(__name__))
 
 
 class NotifyController:
     def __init__(self):
-        self.notify_url = app.config['NOTIFY_SERVICE_URL']
         self.request_password_change_template = app.config['NOTIFY_REQUEST_PASSWORD_CHANGE_TEMPLATE']
         self.confirm_password_change_template = app.config['NOTIFY_CONFIRM_PASSWORD_CHANGE_TEMPLATE']
         self.request_create_account_template = app.config['NOTIFY_REQUEST_CREATE_ACCOUNT_TEMPLATE']
         self.confirm_create_account_template = app.config['NOTIFY_CONFIRM_CREATE_ACCOUNT_TEMPLATE']
+        self.topic_id = app.config['PUBSUB_TOPIC']
+        self.project_id = app.config['GOOGLE_CLOUD_PROJECT']
+        self.publisher = None
 
-    def _send_message(self, email, template_id, personalisation=None, reference=None):
+    def _send_message(self, email, template_id, personalisation):
         """
-        Send message to gov.uk notify wrapper
-        :param email: email address of recipient
-        :param template_id: the template id on gov.uk notify to use
-        :param personalisation: placeholder values in the template
-        :param reference: reference to be generated if not using Notify's id
+        Send message to gov.uk notify via pubsub topic
+        :param email: str email address of recipient
+        :param template_id: the template id on gov.uk notify to be used
+        :param personalisation: personalisation if required
         """
         if not app.config['SEND_EMAIL_TO_GOV_NOTIFY']:
             logger.info("Notification not sent. Notify is disabled.")
             return
 
         notification = {
-            "emailAddress": email,
+            'notify': {
+                'email_address': email,
+                'template_id': template_id,
+                'personalisation': {}
+            }
         }
         if personalisation:
-            notification.update({"personalisation": personalisation})
-        if reference:
```

Comment:
Do we want to delete reference or is that not relevant anymore?

Reply:
I don't think its relevant as I could not find the reference to the attribute being used ever. 
`_send_message` is referred from `request_to_notify` which is referred from the following functions which never passed reference as an attribute.
- `send_confirm_created_email`
- `send_create_account_email`
- `send_confirm_change_email`
- `send_password_change_email`